### ISSUE TITLE
Documentation for DIPper Scripts and Data

### ISSUE THREAD

**Issue by mbrush** [monarch-initiative/dipper#52]
We should define documentation standards for  defining python classes and ingest scripts for specific sources.  

This should include an evolving **specification** document that outlines the desired triples to be produced by a python class or for a specific data source,  to elicit feedback and inform code as it is being written. It will help all developers (data wranglers, coders, ontologists, and UI)  have a common understanding of the data so they can effectively contribute their expertise in an aligned and efficient way.

It should also include **final documentation** that will help inform QA of the data, and production of documentation for DIPper classes and datasets in Scigraph (e.g. a cmap illustrating the data graph produced for a given source). It will also inform development of outward facing documentation will help users to  understand what is in the data, and how to query and navigate it.

**Proposal for Specification Documentation:**
**1 Python Class Documentation:**  A Github ticket for each python class (e.g. genoytpe.py, g2passoc.py) that defines the triple patterns it should generate, including:
- Entity declarations: what objects (instances and classes) in the data are instantiated, what is their asserted rdf:type, and what metadata is captured about them (e.g. descriptions, labels, etc)
- Triples/Statements: what statements are made about or between these entities

	
**2  Source Ingest Script Documentation:** A Github ticket for each data source that defines all types of triples we want to generate in the graph for that source. 
- As with class documentation, this will describe entity declarations and statements about these entities. Additionally, it could define mappings required between source terms/literals and Monarch ontology IRIs could be specified here  
- The documentation can and should reference all python classes called to define most of the triples, and additionally indicate any nuances or unique aspects of the data that requires consideration or additional modeling/triples not defined in the general classes.

**Comments:**
1 We should distinguish entity declarations and triples that are **direct/asserted** (created directly from data at ingest) vs **derived/inferred** (result from post-processing or reasoning)

2 For the initial specification (before coding), triples should be described as best as possible - i.e. use actual term  iris or labels if we know them, and placeholder names for these if these are not yet determined. And in some cases, the model or design patterns for the statements we want to represent will not be clear, so all we can document is a statement in plain English, and this can inform a modeling ticket for GENO.

3 We should define a standard syntax/format for encoding these specifications.  In the initial specification, this should be a informal/loose enough to allow descriptions of classes when specific IRIs are not yet defined. In the final documentation, we can consider using a more formal notations (e.g. N3).
- **Proposal**: Use something similar to disco2ttl YAML syntax for specifications - it is simple/flexible, and something we all know already.

4 When classes and data sources scripts are complete, document the triples it creates in the header (TBD: use term iris and/or a real example with instance data.

5 Nicole/Bryan will create tickets even for completed classes and sources to support documentation, and QA of the modeling and the data in Scigraph.

**Comment by kshefchek** [comment]
I think a good first step would be setting up a documentation pages that are generated from the docstrings in our code.  Maybe use something like Sphinx and host it on github.io?

**Comment by nathandunn** [comment]
We've also had good luck with read the docs.

**Comment by kshefchek** [comment]
Either one, github pages and read the docs are the two I see most commonly used, but I don't know the pros/cons of each.

**Comment by mbrush** [comment]
We have been working to define a concept-map based Standard Graphical Notation for specifying and/or documenting data produced by DIPper scripts.  It is ready for feedback, and documented in the google doc [here.](https://docs.google.com/document/d/1cww1xIKHPClnW7z5X--tKEGKPsT03ORC_qr-AumCDIU/edit#)

**Comment by nlwashington** [comment]
@kshefchek has added docs to readthedocs:
http://dipper.readthedocs.org/en/latest/

how should we include images (such as the cmaps) in this documentation?

**Comment by nlwashington** [comment]
it looks like we can include image references in our documentation, like:
http://markdown-guide.readthedocs.org/en/latest/basics.html#images-inline

the images could certainly live in git, but i would argue that only the pdfs should live there.
